Tests/Realm.Tests/Database/TestObjects.cs

```diff
@@ -177,7 +177,7 @@ public class ListsObject : RealmObject
         public IList<RealmInteger<long>?> NullableInt64CounterList { get; }
     }
 
-    public class SetsObject : RealmObject
+    public class CollectionsObject : RealmObject
```

Comment:
Are we going to use this on all other collection tests too?

Reply:
We'll see, I didn't want to refactor a bunch of other tests as that would explode the size of this PR. But I needed the other collections to be able to write tests that take a native collection as an argument.